File: wapp2/Tasks/Models/Task.cs
using System.ComponentModel.DataAnnotations;

namespace Tasks.Models
{
    public class TaskModel
    {
        public int Id { get; set; }

        [Required, StringLength(150)]
        public string Title { get; set; } = string.Empty;

        [StringLength(50)]
        public string? Category { get; set; }

        [StringLength(1000)]
        public string? Description { get; set; }

        public int? OwnerUserId { get; set; }

        public DateTime? DueDate { get; set; }

        public bool IsCompleted { get; set; }

        [StringLength(20)]
        public string? Priority { get; set; }

        [Required, StringLength(20)]
        public string Status { get; set; } = "pending";

        public DateTime CreatedAt { get; set; }

        public DateTime? UpdatedAt { get; set; }
    }
}

File: full-web-app/src/features/tasks/types/task.ts
export type TaskItem = {
  id: number;
  title: string;
  category: string | null;
  description: string | null;
  ownerUserId: number | null;
  dueDate: string | null;
  isCompleted: boolean;
  priority: string | null;
  status: string;
  createdAt: string;
  updatedAt: string | null;
};

export type TaskPayload = {
  title: string;
  category: string;
  description: string;
  dueDate: string | null;
  isCompleted: boolean;
  priority: string;
  status: string;
};
